src/gt4py/next/ffront/foast_passes/type_alias_replacement.py

```diff
@@ -0,0 +1,99 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2023, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from dataclasses import dataclass
+from typing import Any
+
+import gt4py.next.ffront.field_operator_ast as foast
+from gt4py.eve import NodeTranslator, traits
+from gt4py.eve.concepts import SourceLocation, SymbolName, SymbolRef
+from gt4py.next.ffront import dialect_ast_enums
+from gt4py.next.ffront.fbuiltins import TYPE_BUILTIN_NAMES
+from gt4py.next.type_system import type_specifications as ts
+from gt4py.next.type_system.type_translation import get_scalar_kind
+
+
+@dataclass
+class TypeAliasReplacement(NodeTranslator, traits.VisitorWithSymbolTableTrait):
+    """
+    Replace Type Aliases with their actual type.
+
+    After this pass, the type aliases used for explicit construction of literal
+    values and for casting field values are replaced by their actual types.
+    """
+
+    closure_vars: dict[str, Any]
+
+    @classmethod
+    def apply(
+        cls, node: foast.FunctionDefinition | foast.FieldOperator, closure_vars: dict[str, Any]
+    ) -> tuple[foast.FunctionDefinition, dict[str, Any]]:
+        foast_node = cls(closure_vars=closure_vars).visit(node)
+        closures_copy = closure_vars.copy()
+        for key, value in closures_copy.items():
+            if isinstance(value, type) and key not in TYPE_BUILTIN_NAMES:
+                closure_vars[value.__name__] = closure_vars.pop(key)
+
+        return foast_node, closure_vars
+
+    def _get_actual_type_name(
+        self, node_id: SymbolName | SymbolRef
+    ) -> str | SymbolName | SymbolRef:
+        if (
+            node_id in self.closure_vars
+            and isinstance(self.closure_vars[node_id], type)
+            and node_id not in TYPE_BUILTIN_NAMES
+        ):
+            return self.closure_vars[node_id].__name__
+        return node_id
+
+    def visit_Name(self, node: foast.Name, **kwargs) -> foast.Name:
+        actual_type_name = self._get_actual_type_name(node.id)
+        new_node_id = actual_type_name if not isinstance(actual_type_name, SymbolRef) else node.id
```

Comment:
Same reasoning as below.
```suggestion
        if self.is_type_alias(node.id):
            return self.closure_vars[node_id].__name__
        return node
```

Reply:
The suggestation has been modified to return the node in the condition:
```
return foast.Name(
                id=self.closure_vars[node.id].__name__, location=node.location, type=node.type
            )
```